test/unit/lib/blob/blob.c/blob_ut.c

```diff
@@ -7394,6 +7395,127 @@ blob_seek_io_unit(void)
 	ut_blob_close_and_delete(bs, blob);
 }
 
+static void
+bs_dev_io_complete_cb(struct spdk_io_channel *channel, void *cb_arg, int bserrno)
+{
+	g_bserrno = bserrno;
+}
+
+static void
+blob_shallow_copy(void)
+{
+	struct spdk_blob_store	*bs = g_bs;
+	struct spdk_blob_opts	blob_opts;
+	struct spdk_blob		*blob;
+	spdk_blob_id			blobid;
+	uint64_t				num_clusters = 4;
```

Comment:
Why 4?

Reply:
4 to have a couple of cluster allocated per snapshot interleaved with non allocated clusters.
If you follow the operations it is clearer:
- Write on cluster 2 and 4 of blob
- Make a snapshot over blob
- Write on cluster 1 and 3 of blob
- Make a shallow copy of blob over bdev
The shallow copy must copy only cluster 1 and 3 because 2 and 4 are allocated in the old snapshot